src/p2p/net_node.inl

```diff
@@ -1095,19 +1161,22 @@ namespace nodetool
     {
       if(conn_count < expected_white_connections)
       {
-        //start from white list
-        if(!make_expected_connections_count(true, expected_white_connections))
+        //start from anchor list
+        if(!make_expected_connections_count(anchor, P2P_DEFAULT_ANCHOR_CONNECTIONS_COUNT))
           return false;
-        //and then do grey list
-        if(!make_expected_connections_count(false, m_config.m_net_config.connections_count))
+        //then do white list
+        if(!make_expected_connections_count(white, expected_white_connections))
```

Comment:
Having re-read the paper I agree with @moneromooo-monero, but this could be a nice little thing to test. Thinking about it though, going from potential list of 1500 peers after say 3 days uptime back down to just 8 at restart seems like a rather extreme swing - but I guess it means your attacker has to successfully fill most of your peer list in order to have the chance of 2/1500 of your randomly selected restart list belonging to them.